From 89d15f4db60294cc9ce24556ddf3e370db01430d Mon Sep 17 00:00:00 2001
From: Jan Pavlinec <jan.pavlinec@nic.cz>
Date: Fri, 27 Nov 2020 14:49:43 +0100
Subject: [PATCH] umdns: add check for seccomp list

This should fix  an issue when user have a
router with enabled seccomp and tries to
run umdns package which was build with
SDK with disabled seccomp support.

Signed-off-by: Jan Pavlinec <jan.pavlinec@nic.cz>
---
 package/network/services/umdns/Makefile         | 2 +-
 package/network/services/umdns/files/umdns.init | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/package/network/services/umdns/Makefile b/package/network/services/umdns/Makefile
index dc032af31b4e..41945ce0079a 100644
--- a/package/network/services/umdns/Makefile
+++ b/package/network/services/umdns/Makefile
@@ -8,7 +8,7 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=umdns
-PKG_RELEASE:=1
+PKG_RELEASE:=2
 
 PKG_SOURCE_URL=$(PROJECT_GIT)/project/mdnsd.git
 PKG_SOURCE_PROTO:=git
diff --git a/package/network/services/umdns/files/umdns.init b/package/network/services/umdns/files/umdns.init
index c0b8cedd79b5..1ae1cccb039c 100644
--- a/package/network/services/umdns/files/umdns.init
+++ b/package/network/services/umdns/files/umdns.init
@@ -33,7 +33,7 @@ start_service() {
 
 	procd_open_instance
 	procd_set_param command "$PROG"
-	procd_set_param seccomp /etc/seccomp/umdns.json
+	[ -f /etc/seccomp/umdns.json ] && procd_set_param seccomp /etc/seccomp/umdns.json
 	procd_set_param respawn
 	procd_open_trigger
 	procd_add_config_trigger "config.change" "umdns" /etc/init.d/umdns reload
